src/main/scala/jio/package.scala

```diff
@@ -50,7 +51,11 @@ package object jio extends JioFiles {
       Files.setLastModifiedTime(path, jnfa.FileTime.from(nanoSeconds, TimeUnit.NANOSECONDS))
 
     def tryLock():jnc.FileLock     = withWriteChannel(_.tryLock)
-    def truncate(size: Long): Unit = withWriteChannel(_ truncate size )
+    def truncate(size: Long): Unit = withWriteChannel {
+      case c if c.size > size => c truncate size
+      case c if c.size < size => c write (ByteBuffer wrap Array.fill((size - c.size).toInt)(0.toByte), c.size)
+      case _                  => // sizes are equal
```

Comment:
And having a method called "truncate" which expands the file arbitrarily large... but I guess if we're really going for the identity filesystem those are the required semantics.